include/RAJA/policy/cuda/params/reduce.hpp

```diff
@@ -8,37 +8,86 @@
 #include "RAJA/policy/cuda/reduce.hpp"
 #include "RAJA/pattern/params/reducer.hpp"
 
+#include "RAJA/policy/cuda/policy.hpp"
+
 namespace RAJA {
 namespace expt {
 namespace detail {
 
   // Init
-  template<typename EXEC_POL, typename OP, typename T>
+  template<typename EXEC_POL, template <typename, typename, typename> class OP, typename T, typename D, typename I, typename VType>
+  camp::concepts::enable_if< type_traits::is_cuda_policy<EXEC_POL> >
+  init(Reducer<OP, T, VType, D, I, true>& red, RAJA::cuda::detail::cudaInfo& ci)
```

Comment:
These seem much the same as below, could these be de-duplicated and the places where they differ be done in member functions of Reducer instead?

Reply:
Along with the partial specialization of `Reducer`, the `set()` function within `Reducer` looks the same for both basic data types and `ValLoc`. Duplicate versions of init/combine/resolve are no longer needed.